infrastructure/src/main/java/org/corfudb/infrastructure/RemoteMonitoringService.java

```diff
@@ -727,12 +740,9 @@ private void correctWrongEpochs(PollReport pollReport) {
      *
      * @param layoutCompletableFutureMap Map of layout server endpoints to their layout requests.
      */
-    private void updateTrailingLayoutServers(Map<String, CompletableFuture<Layout>> layoutCompletableFutureMap) {
+    private void updateTrailingLayoutServers(
+            Layout latestLayout, Map<String, CompletableFuture<Layout>> layoutCompletableFutureMap) {
 
-        // We should utilize only the unmodified management layout as it has already been committed to the layout
```

Comment:
Please take a note of the comments added here.
We have introduced this bug earlier too and we should be extra cautious.
This layout is being modified by the sealing method in `correctWrongEpochs` and cannot be used to update the layout servers directly. We need an unmodified copy.

Reply:
This code is identical to the previous version, it uses the same instance of a management layout like in the previous version. 
If we want to change this logic and use an unmodified copy of the layout then let's create a pr for that.
Not sure if we want to fix the bug in this pr.

Reply:
The layout we are referring to is being modified in correctWrongEpoch. This is using the same layout.
We already fixed this bug in this PR #1755

Reply:
I can't get your question, do you believe there is a bug in this pr or not?

Reply:
So, I double checked your PR and this change, don't see any issues. Please explain if you see any issues here.

Reply:
I am saying that **this PR** is introducing the same bug which was fixed by 1755.
The correctWrongEpochs is modifying the layout and that should not be used to update the layout servers.

Reply:
If you take a look you can see that we send a modified version of the management layout, we don't use the old version of the management layout. So, the changes didn't bring new bugs and it's the same as a previous version. We just pass the layout instead of getting it from serverContext

Reply:
And also it seems to me that your #1755 doesn't fix anything. Correct me if I am wrong

Reply:
Ok, now I see what you are trying to say. It's really dangerous and completely unclear part of the code.

Fixed.

Reply:
@zalokhan if you are ok with the changes, let's have progress, or leave your comments